src/tbp/monty/frameworks/models/states.py

```diff
@@ -217,45 +221,56 @@ class GoalState(State):
     Used by goal-state generators (GSGs) to communicate goal states to other GSGs, and
     to motor actuators.
 
-    The state variables generally have the same meanign as for the base State
+    The state variables generally have the same meaning as for the base :class:`State`
     class, and they represent the target values for the receiving system. Thus
     if a goal-state specifies a particular object ID (non-morphological feature)
     in a particular pose (location and morphological features), then the receiving
     system should attempt to achieve that state.
 
     Note however that for the goal-state, the confidence corresponds to the conviction
-        with which a GSG believes that the current goal-state should be acted upon.
-        Float bound in [0,1.0].
+    with which a GSG believes that the current goal-state should be acted upon. Float
+    bound in [0.0, 1.0].
     """
 
     def __init__(
         self,
-        location,
-        morphological_features,
-        non_morphological_features,
-        confidence,
-        use_state,
-        sender_id,
-        sender_type,
-        goal_tolerances,
+        location: np.ndarray,
```

Comment:
I believe we need something like the below to allow for `location` and `morphological_features` to be `None`:

```
        location: Optional[np.ndarray],
        morphological_features: Optional[Dict[str, Any]],
```